### DISCUSSION TITLE
startproject composeexample

### DISCUSSION THREAD

**Discussion by eddhyne** [community/community#64497] [Programming Help]
### Body

I am trying to get the Docker sample code Quickstart: Compose and Django at
https://github.com/docker/awesome-compose/blob/master/official-documentation-samples/django/

When I run "docker compose run web django-admin startproject composeexample ."
it does not create the composeexample directory. Please help.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Kaisama** [comment]
I'd be happy to help you troubleshoot the issue. If the docker compose run web django-admin start project compose example . Command is not creating the compose example directory as expected, here are a few steps you can take to troubleshoot the problem:

1-Check Docker Compose Version: Make sure you are using a version of Docker Compose that supports the docker compose syntax. As of my last update in September 2021, Docker Compose was transitioning to using the docker compose command instead of docker-compose, so ensure you are using a version that aligns with this change.

2-Check Working Directory: Make sure you are running the command from the correct working directory. The docker compose command is sensitive to the working directory you're in when you run the command. The command you provided should be run within the root directory of the repository where the docker-compose.yml file is located.

3-Check Docker Compose File: Ensure that the docker-compose.yml file in the repository is correctly configured. It should define the necessary services, volumes, and other configurations required to set up the Django project. If the configuration is incorrect, it could lead to unexpected behavior.

4-Check for Errors: After running the command, carefully review the output for any error messages or warnings. This could provide clues as to what might be going wrong.

5-Alternative Approach: If you continue to experience issues, you might try an alternative approach to starting the project. You can manually run a Docker container with the Django image and then run the django-admin startproject command within the container. Here's an example of how you can do that:

docker run -v $(pwd):/app django django-admin startproject composeexample .

This command runs a Django container and mounts the current directory as /app inside the container. The django-admin startproject command is then executed within the container.

6-GitHub Issue: If none of the above solutions work, consider checking the GitHub repository's issues page. There might be others who have encountered similar problems and found solutions or workarounds.

Remember that the software landscape can change quickly, so it's possible that there have been updates or changes since my last knowledge update in September 2021. Always refer to the official documentation and resources for the most up-to-date information.

**Reply by eddhyne** [reply]
I have tried your suggestions and I still can't get it to create composeexample.
I have Docker Compose version v2.20.2-desktop.1
I am getting the following:
PS Z:\Docker+Django\Docker download from Docker Hub\Ed's Code> docker compose run web django-admin startproject composeexample .
[+] Creating 1/0
 ✔ Container edscode-db-1  Created                                                                                                                                                                                          0.0s 
[+] Running 1/1
 ✔ Container edscode-db-1  Started                                                                                                                                                                                          0.4s 
CommandError: /code/manage.py already exists. Overlaying a project into an existing directory won't replace conflicting files.
PS Z:\Docker+Django\Docker download from Docker Hub\Ed's Code> docker run -v $(pwd):/app django django-admin startproject composeexample .
docker: invalid reference format.
See 'docker run --help'

I am running Windows 11 Enterprise
See my attachment
![Snag_f6ea8b6](https://github.com/community/community/assets/92999884/5b0f18f0-aa18-4b5c-9533-d29cbf0622fd)
[DockErrors.docx](https://github.com/community/community/files/12397903/DockErrors.docx)

**Comment by Kaisama** [comment]
1- docker: invalid reference format: This error is occurring because of the way you're using the -v flag to mount the current directory into the Docker container. The syntax $(pwd) is valid in Unix-like shells like Bash, but it seems like you're using PowerShell on Windows, where the syntax is different.

Here's how you can correct the command:

docker run -v ${PWD}:/app django django-admin startproject composeexample .

${PWD} is the equivalent of $(pwd) in PowerShell. It represents the current working directory.

-v ${PWD}:/app mounts the current directory on your host machine into the /app directory in the Docker container.

django specifies the Docker image you want to use.

django-admin startproject composeexample . is the command you want to run inside the Docker container. It will create a new Django project named composeexample in the /app directory.

**Comment by eddhyne** [comment]
Thanks very much. I just followed what was in the awesome-compose / official-documentation-samples/django/  document online.
awesome-compose/official-documentation-samples/django at master · docker/awesome-compose (github.com)<https://github.com/docker/awesome-compose/tree/master/official-documentation-samples/django/>
Someone should fix that or add how it should be done in Windows like you did. Thanks again.

Ed Dhyne
Direct: 954-608-6603
***@***.******@***.***>
www.linkedin.com
